src/execution_plan/ops/op_merge.c

```diff
@@ -185,18 +187,23 @@ static OpResult MergeInit(OpBase *opBase) {
 
 static Record _handoff(OpMerge *op) {
 	Record r = NULL;
-	if(array_len(op->output_records)) r = array_pop(op->output_records);
+	if(op->matched_records) {
+		if(array_len(op->matched_records)) r = array_pop(op->matched_records);
+		else if(op->created_records) {
```

Comment:
Misleading indentation, should be the same as `if(op->matched_records) {`
* my understanding, `matched_records` contains all records that were matched.
`created_records` contains all records that were created.

Reply:
fixed